### ISSUE TITLE
Blurred image when setting output = direct3d

### ISSUE THREAD

**Issue by maximus105** [joncampbell123/dosbox-x#1732]
**Describe the bug**
Setting output = direct3d produces a blurred image with scaler = advinterp2x forced and also all other scaler values. At the same time, in DOSBox SVN-Daum, when setting output = direct3d and when scaler = advinterp2x forced, the image is clear and more correct.

**Environment (please complete the following information):**
 - Windows 10 x64 2004
 - DOSBox-X 0.83.3 x64 SDL1

[sdl]
fullscreen = true
fullresolution = desktop
output = direct3d

[render]
aspect = true
doublescan = false
scaler = advinterp2x forced
autofit = true

[dosbox]
vmemsize = 8
memsize = 32

**Comment by maximus105** [comment]
Here the bug is in the render output = direct3d. With the same scaler = advinterp2x forced value in the output = openglnb render, the image is clear, correct and looks without bugs. In other versions of DOSBox Staging v0.75.0, DOSBox ECE, DOSBox SVN-Daum with output = direct3d or ddraw, scaler = advinterp2x forced, the image is as clear as in DOSBox-X with output = openglnb, scaler = advinterp2x forced. But the opengl render does not allow vsync to be used, as stuttering sounds and music in games appear with it. At the same time, the direct3d render does not have these bugs with vsync, but there are bugs using scaler = advinterp2x forced and other scaler values.

**Comment by Wengier** [comment]
DOSBox ECE or DOSBox-staging 0.75 does not support "output = direct3d" at all - it will simply be ignored if you specify this. DOSBox ECE does support "output = ddraw" though, but it is something different.

**Comment by maximus105** [comment]
@Wengier I know about it. I say that DOSBox SVN-Daum in output = direct3d and scaler = advinterp2x forced mode shows a really high-quality picture without soap, exactly the same as DOSBox-X with output = openglnb. The question is that DOSBox-X with output = direct3d and scaler = advinterp2x forced very lathers the picture. This is exactly the bug.

**Comment by Wengier** [comment]
@maximus105 I have tried using the exact same config file as you posted, and it does not give a bad or blurry picture when I run DOSBox-X on my computer. Perhaps scaler=advinterp2x is not fully supported on your computer, and thus "forced" will not work as it is expected to.

**Comment by maximus105** [comment]
Compare screenshots from DOSBox SVN-Daum and DOSBox-X with the same parameters output = direct3d and scaler = advinterp2x forced.

Screenshots from DOSBox SVN-Daum:
![dosbox_1](https://user-images.githubusercontent.com/58270614/87697247-88121980-c79a-11ea-975d-2c3693d7e944.png)
![dosbox_2](https://user-images.githubusercontent.com/58270614/87697252-89dbdd00-c79a-11ea-8c40-2b03f92ad0c6.png)

Screenshots from DOSBox-X:
![dosbox-x_1](https://user-images.githubusercontent.com/58270614/87697285-95c79f00-c79a-11ea-999a-9a7e312f4e75.png)
![dosbox-x_2](https://user-images.githubusercontent.com/58270614/87697291-96f8cc00-c79a-11ea-8e3a-088a7f9ef0d6.png)

**Comment by Wengier** [comment]
@maximus105 The screenshots look almost identical to my eyes. The differences are very minor if they exist at all.

**Comment by maximus105** [comment]
@Wengier Do not forget to click “enlarge” on the screenshot so that the original resolution of the screenshot is. The differences are very significant. On all faces of objects in DOSBox-X, a blurry image (soap image) is visible.

**Comment by kcgen** [comment]
Depending on the GPU driver and its setting, it might be adding it own Gaussian-blending when hardware-scaling a smaller image to some larger resolution. Plus, to "convolute" matters even more;-), web-browsers will gaussian-blur pixels when you zoom beyond 1:1; so the differences viewed in a browser will still appear less pronounced.

![2020-07-16_10-49](https://user-images.githubusercontent.com/1557255/87705021-1f9f4a00-c752-11ea-95f2-c2a5c0cfa8e3.png)

![2020-07-16_10-49_1](https://user-images.githubusercontent.com/1557255/87705024-2037e080-c752-11ea-8b88-3c7abaf698f8.png)

**Comment by Wengier** [comment]
@kcgen Thanks for the information, and I think it is a good thing that you are so familiar with stuff like Gaussian-blending.

**Comment by kcgen** [comment]
@Wengier , I'm not sure where it's happening though :disappointed:  It's tough being a square pixel, when every layer of software wants to blend up-scaled pixels!

**Comment by maximus105** [comment]
I took screenshots with a third-party program (MSI Afterburner) in PNG without loss, so what I see on the monitor is the same in these screenshots. Therefore, I see the difference in DOSBox SVN-Daum and DOSBox-X. Is it possible to somehow fix this bug and get the same clear image in DOSBox-X as in DOSBox SVN-Daum with output = direct3d and scaler = advinterp2x forced with fullscreen = true?

**Comment by maximus105** [comment]
With the same parameters:
[sdl]
fullscreen = true
fullresolution = desktop
output = direct3d

[render]
aspect = true
doublescan = false
scaler = advinterp2x forced
autofit = true

in SDL2 the image raster is not displayed full screen (it is small in the center of the screen), while in SDL1 the image is correct and full screen. (with output = openglnb the image raster is also correct and full screen in SDL2). It looks like this is another bug with SDL2. Maybe there is something to do with this blurry image in direct3d?

**Comment by maximus105** [comment]
In DOSBox-X 0.83.5 x64 SDL1, after opening the mapper and then closing it, the screen is damaged (stretched vertically so that the top and bottom images go off the screen).

Environment (please complete the following information):

 - Windows 10 x64 1909
 - DOSBox-X 0.83.5 x64 SDL1

[sdl]
fullscreen = true
fullresolution = desktop
output = default

[render]
aspect = true
doublescan = false
scaler = advinterp2x forced
autofit = true

[dosbox]
vmemsize = 8
memsize = 32

**Comment by Wengier** [comment]
@maximus105 Did the issue occur in the previous version? Also, are you using the MinGW SDL1 release (found in the release page) or the Visual Studio SDL1 release?

**Comment by maximus105** [comment]
@Wengier Yes. This bug was back in version 0.83.4. In version 0.83.4 I used build SDL1 from here:
https://github.com/joncampbell123/dosbox-x/releases/download/dosbox-x-v0.83.4/dosbox-x-windows-20200803-055400-windows.zip

And in version 0.83.5 from here:
https://github.com/joncampbell123/dosbox-x/releases/download/dosbox-x-v0.83.5/dosbox-x-mingw-win64-20200901093925.zip

**Comment by Wengier** [comment]
@maximus105 I do not use full-screen mode a lot (I cannot access the menu bar in this mode for example) - it looks like opening the mapper editor will switch to window mode if it is in full-screen mode, and closing the mapper editor will return to full-screen mode. Seem to return to full-screen mode fine at least on this computer, so it may be specific to configuration. Have you tried SDL2 version to see how it works?

**Comment by maximus105** [comment]
@Wengier How to try using maximized windowed mode instead of fullscreen mode for SDL1 version?
I tried SDL2 version. Everything is fine in it when starting the mapper and then exiting it, but in the SDL2 version there is a blurry image when using fastbioslogo = true. This is detailed here: https://github.com/joncampbell123/dosbox-x/issues/1523

**Comment by Wengier** [comment]
@maximus105 When you open DOSBox-X in window mode, there is a "Maximize" button (next to the "Close" button), correct? Just click it, then the DOSBox-X window will be maximized to fit the screen. Compared with full-screen mode, it has the title bar and the menu bar visible, which may be very useful.

I have tried the "Reboot Guest system" button in CGA and EGA modes, and it seems to work here. Not sure why it freezes on your system.

**Comment by maximus105** [comment]
Here is my CGA config with which DOSBox-X freezes when clicking "Reboot Guest system":
[CGA_Mono_White.zip](https://github.com/joncampbell123/dosbox-x/files/5180092/CGA_Mono_White.zip)

**Comment by Wengier** [comment]
@maximus105 Thanks for uploading the config file. I think the freeze only occurs with the command "keyb ru" added in CGA/EGA modes, and I believe it is already fixed in the latest code.
